msal/src/Microsoft.Identity.Client/UIParent.cs

```diff
@@ -49,15 +49,53 @@ public UIParent()
             CoreUIParent = new CoreUIParent();
         }
 
+#if iOS
+        public UIParent(bool useEmbeddedWebview)
+        {
+            CoreUIParent.UseEmbeddedWebview = useEmbeddedWebview;
+        }
+#endif
+
 #if ANDROID
+        private Activity Activity { get; set; }
+
+        private static readonly string[] _chromePackages =
+        {"com.android.chrome", "com.chrome.beta", "com.chrome.dev"};
 
         /// <summary>
         /// Initializes an instance for a provided activity.
         /// </summary>
         /// <param name="activity">parent activity for the call. REQUIRED.</param>
         public UIParent(Activity activity)
         {
-            CoreUIParent = new CoreUIParent(activity);
+            Activity = activity;
+            CoreUIParent = new CoreUIParent(Activity);
+        }
+
+        public UIParent(Activity activity, bool useEmbeddedWebview) :this(activity)
+        {
+            CoreUIParent.UseEmbeddedWebview = useEmbeddedWebview;
+        }
+
+        public static bool IsSystemWebviewAvailable()
+        {
+            PackageManager packageManager = Application.Context.PackageManager;
+
+            string installedChromePackage = null;
+            try
+            {
+                for (int i = 0; i < _chromePackages.Length; i++)
+                {
+                    packageManager.GetPackageInfo(_chromePackages[i], PackageInfoFlags.Activities);
+                    installedChromePackage = _chromePackages[i];
+                }
+            }
+            catch (PackageManager.NameNotFoundException)
+            {
+                return false;
+            }
```

Comment:
is it required that all packages are available to say system webview is available or is one package enough? if one is enough, then I'm not sure this code is doing what is intended

Reply:
we were using something similar in [AuthenticationActivity](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/blob/dev/core/src/Platforms/Android/SystemWebview/AuthenticationActivity.cs#L225), but you are right, was doing too much here...let me know what you think with the changes.

Reply:
not really sure this is all needed. let's talk in person about this one as a follow up item

---
In reply to: [188810966](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1055#discussion_r188810966) [](ancestors = 188810966)

Reply:
Sounds good.

---
In reply to: [189028256](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1055#discussion_r189028256) [](ancestors = 189028256,188810966)